src/otx/algo/instance_segmentation/maskrcnn_tv.py

```diff
@@ -0,0 +1,265 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""TV MaskRCNN model implementations."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import torch
+from torch import nn
+from torchvision import tv_tensors
+from torchvision.models.detection.backbone_utils import _resnet_fpn_extractor, _validate_trainable_layers
+from torchvision.models.detection.faster_rcnn import FastRCNNConvFCHead, FastRCNNPredictor, RPNHead, _default_anchorgen
+from torchvision.models.detection.mask_rcnn import MaskRCNN_ResNet50_FPN_V2_Weights, MaskRCNNHeads, MaskRCNNPredictor
+from torchvision.models.resnet import resnet50
+
+from otx.algo.instance_segmentation.torchvision.maskrcnn import TVMaskRCNN
+from otx.algo.instance_segmentation.torchvision.roi_head import OTXTVRoIHeads
+from otx.core.config.data import TileConfig
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.instance_segmentation import InstanceSegBatchDataEntity, InstanceSegBatchPredEntity
+from otx.core.data.entity.utils import stack_batch
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.metrics.mean_ap import MaskRLEMeanAPCallable
+from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
+from otx.core.model.instance_segmentation import ExplainableOTXInstanceSegModel
+from otx.core.schedulers import LRSchedulerListCallable
+from otx.core.types.label import LabelInfoTypes
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+    from torch.nn.modules import Module
+
+    from otx.core.metrics import MetricCallable
+
+
+class TVMaskRCNNR50(ExplainableOTXInstanceSegModel):
+    """Torchvision MaskRCNN model with ResNet50 backbone."""
+
+    def __init__(
+        self,
+        label_info: LabelInfoTypes,
+        optimizer: OptimizerCallable = DefaultOptimizerCallable,
+        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
+        metric: MetricCallable = MaskRLEMeanAPCallable,
+        torch_compile: bool = False,
+        tile_config: TileConfig = TileConfig(enable_tiler=False),
+    ) -> None:
+        super().__init__(
+            label_info=label_info,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+            tile_config=tile_config,
+        )
+        self.image_size = (1, 3, 1024, 1024)
```

Comment:
We need to think how we can avoid hardcoded numbers in the models. Image_size is something that user may want to change. Same for mean/std

Reply:
I agree, it's only used for export, and currently, there's no other way to get attributes from recipe configs. We could find a way to access recipe parameters, but I don't know how to do it. This should be addressed in a different PR.

Reply:
I will put this task to our backlog